tools/check_version.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+var {spawnSync} = require('child_process');
+
+const expected = process.env.npm_package_version;
+
+let contents;
+
+// yarn_install in user WORKSPACE
+let maybe = path.resolve(process.cwd(), '../../../../build_bazel_rules_typescript/package.bzl');
+if (fs.existsSync(maybe)) {
+  contents = fs.readFileSync(maybe, 'utf8');
+}
+
+// local yarn call in user WORKSPACE
+if (!contents && process.cwd().endsWith('node_modules/@bazel/typescript')) {
+  // no version check
+  process.exit(1);
+}
+
+// yarn_install in build_bazel_rules_typescript WORKSPACE
+maybe = path.resolve(process.cwd(), '../build_bazel_rules_typescript/package.bzl');
+if (fs.existsSync(maybe)) {
+  contents = fs.readFileSync(maybe, 'utf8');
+}
+
+// local yarn call in build_bazel_rules_typescript WORKSPACE
+maybe = path.resolve(process.cwd(), 'package.bzl');
+if (fs.existsSync(maybe)) {
+  contents = fs.readFileSync(maybe, 'utf8');
+}
+
+if (!contents) {
```

Comment:
logic might be easier to read if we comment about which failure mode we are checking for at each step

Reply:
cleaned this code up now that we have separate packages and check_version.js is not called in the main package.json anymore